erpnext/controllers/accounts_controller.py

```diff
@@ -829,10 +829,10 @@ def validate_currency(self):
 				party_account_currency = get_party_account_currency(party_type, party, self.company)
 
 				if (party_account_currency
-						and party_account_currency != self.company_currency
-						and self.currency != party_account_currency):
-					frappe.throw(_("Accounting Entry for {0}: {1} can only be made in currency: {2}")
-								 .format(party_type, party, party_account_currency), InvalidCurrency)
+						and (self.currency != party_account_currency
+						and self.currency != self.company_currency)):
+					frappe.throw(_("Accounting Entry for {0}: {1} can only be made in currency: {2} or {3}")
```

Comment:
The earlier message was correct, a party cannot use two currencies at a time, this is only possible for the first transaction.